File: week-01/day-02/algo-factorial/factorial.py
#Iterative factorial solution
def factorial_iterative(input_number):
    result = 1 #Holds final result of factorial

    #Multiply result by current inputNumber val until inputNumber <= 0
    while input_number > 1:
        result *= input_number
        input_number -= 1
    
    return result

#Recursive factorial solution
def factorial_recursive(input_number):
    return 1 if input_number <= 1 else factorial_recursive(input_number - 1) * input_number

File: week-01/day-02/algo-factorial/factorial.js
//Iterative factorial solution
exports.factorialIterative = function(inputNumber) {
    let result = 1n; //Holds final result of factorial

    //Multiply result by current inputNumber val until inputNumber <= 0
    while (inputNumber > 1) {
        result *= BigInt(inputNumber);
        inputNumber--;
    }

    return parseInt(result.toString(), 10);
}

//Recursive factorial solution
exports.factorialRecursive = factorialRecursive;
function factorialRecursive(inputNumber) {
    return inputNumber <= 1 ? 1 : factorialRecursive(inputNumber - 1) * inputNumber;
}